htdocs/web_portal/controllers/user/view_user.php

```diff
@@ -79,9 +79,11 @@ function view_user() {
                     $r->setDecoratorObject('GOCDB ADMIN');
                 }
             }
+            $params['viewingSelf'] = false;
     } else {
         // current user is viewing their own roles, so they can revoke their own roles
         $r->setDecoratorObject('[Self revoke own role]');
+        $params['viewingSelf'] = true;
```

Comment:
I would separate this out from the block of code that determines if the user can revoke roles of the user they are looking at.

Reply:
Yes, that's fixed some of the weirdness!